main.js

```diff
@@ -0,0 +1,169 @@
+
+    return word.split("").reverse().join("");
+}
+console.log(reverse("caterpillar"))
+console.log(reverse("hello"))
+
+
+//   <li>Create a function that takes an array of words and combines them with a dash
+//   <br>
+//     <code > addDashes(['test1', 'test2', 'test3']) // should return "test1-test2-test3"</code>
+//   </li>
+
+const addDashes = function(arr) {
+
+    return arr.join("-");
+}
+console.log(addDashes(['test1', 'test2', 'test3']))
+
+//   <li>Function that will count up to a number and back down and return a string of the climb
+//   <br>
+//     <code> countUpAndDown(3) // should return "1 2 3 2 1"</code>
+//   </li>
+
+const countUpAndDown = function(num) {
+    const arr = []
+    for (let i = 1; i <= num; i++) {
+        arr.push(i)
+    }
+    for (let i = num - 1; i > 0; i--) {
+        arr.push(i)
+    }
+    return arr
+}
+
+console.log(countUpAndDown(3))
+console.log(countUpAndDown(7))
+
+
+//   <li>Write a function that will tell you all of the words in an array that contain the letter `a`
+//   <br>
+//     <code > wordsWithA(['cat', 'rabbit', 'dog', 'frog']) // should return ['cat', 'rabbit']</code>
+//   </li>
+
+const wordsWithA = function(arr) {
+    const wordA = []
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].indexOf('a') != -1) {
+            wordA.push(arr[i]);
+        }
+    }
+    return wordA;
+
+}
+console.log(wordsWithA(['cat', 'rabbit', 'dog', 'frog']))
+
+
+//   <li>Write a function that will tell you all of the words in an array that contain a specified letter
+//   <br>
+//     <code > wordsWithLetter("g", ['cat', 'rabbit', 'dog', 'frog']) // should return ['dog', 'frog']</code>
+//   </li>
+
+const wordsWithLetter = function(letter, arr) {
+    const word = []
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].indexOf(letter) != -1) {
+            word.push(arr[i]);
+        }
+    }
+    return word;
+
+}
+console.log(wordsWithLetter('g', ['cat', 'rabbit', 'dog', 'frog']))
+console.log(wordsWithLetter('r', ['cat', 'rabbit', 'dog', 'frog', 'turtle']))
+
+
+//   <li>Function that returns the longest word in sentence
+//   <br>
+//     <code>longestWord("The cat in the house") // should return "house"</code>
+//   </li>
+
+const longestWord = function(string) {
+    wordArr = string.split(' ')
+    let count = '';
+
+    for (let i = 0; i < wordArr.length; i++) {
+        if (count.length <= wordArr[i].length) {
+            count = wordArr[i];
+        }
+    }
+    return count;
+}
+
+console.log(longestWord("The cat in the house"))
+console.log(longestWord("the homework is soo long"))
+
+
+//   <li>Function that returns the largest even number
+//   <br>
+//     <code>largestEvenNumber([1,2,3,10,4,7,0]) // should return "10"</code>
+//   </li>
+// </ol>
+
+const largestEvenNumber = function(num) {
+
+    let count = 0;
+    for (let i = 0; i < num.length; i++) {
+        if (count <= num[i]) {
+            count = num[i];
+        }
+    }
+    return count;
+}
+

```

Comment:
this question wants to know the largest even number in a given array of numbers.
this is the condition that should iterate through the numbers list 
```
if ( !(nums[i] % 2 )&& nums[i] > largest ){
            largest = nums[i];
        }
```